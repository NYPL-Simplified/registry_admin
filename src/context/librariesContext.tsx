import React, {
  createContext,
  Dispatch,
  useContext,
  useEffect,
  useState,
} from 'react';
export interface LibraryData {
  areas: {
    focus: string[];
    service: string[];
  };
  basic_info: {
    description: string;
    internal_urn: string;
    name: string;
    number_of_patrons: string;
    online_registration: string;
    pls_id: string;
    short_name: string;
    timestamp: string;
  };
  stages: {
    library_stage: 'testing' | 'production' | 'cancelled';
    registry_stage: 'testing' | 'production' | 'cancelled';
  };
  urls_and_contact: {
    authentication_url: string;
    contact_email: string | null;
    contact_validated: string;
    copyright_email: string | null;
    copyright_validated: string;
    help_email: string | null;
    help_validated: string;
    opds_url: string;
    web_url: string;
  };
  uuid: string;
}

type LibrariesContextValues = {
  libraries: LibraryData[];
  setLibraries: Dispatch<React.SetStateAction<LibraryData[]>>;
  setUpdatedLibrary: Dispatch<React.SetStateAction<LibraryData | undefined>>;
};

type LibrariesProviderProps = {
  children: React.ReactNode;
};

export const LibrariesContext = createContext<
  LibrariesContextValues | undefined
>(undefined);

export const LibrariesProvider = ({ children }: LibrariesProviderProps) => {
  const [updatedLibrary, setUpdatedLibrary] = useState<
    LibraryData | undefined
  >();
  const [libraries, setLibraries] = useState<LibraryData[]>([]);

  useEffect(() => {
    if (updatedLibrary) {
      // Find the index of the library with the same uuid as the one
      // that was recently updated.
      const indexOfElementToReplace = libraries.findIndex(
        (library) => library.uuid === updatedLibrary.uuid
      );

      // Replace the library at that index with the updatedLibrary.
      const updatedLibraries = libraries.slice();
      updatedLibraries.splice(indexOfElementToReplace, 1, updatedLibrary);

      // Set libraries to the updated list.
      setLibraries(updatedLibraries);
    }
  }, [updatedLibrary]);

  return (
    <LibrariesContext.Provider
      value={{ libraries, setLibraries, setUpdatedLibrary }}
    >
      {children}
    </LibrariesContext.Provider>
  );
};

const useLibrariesContext = () => {
  const context = useContext(LibrariesContext);
  if (typeof context === 'undefined') {
    throw new Error('useLibraries must be used within a LibraryProvider');
  }
  return context;
};

export default useLibrariesContext;
