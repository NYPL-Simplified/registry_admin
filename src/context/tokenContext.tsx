import React, { createContext, Dispatch, useState } from 'react';

type TokenContextValues = {
  accessToken: string;
  setAccessToken: Dispatch<React.SetStateAction<string>>;
};

type TokenProviderProps = {
  children: React.ReactNode;
};

export const TokenContext = createContext<TokenContextValues | undefined>(
  undefined
);

export const TokenProvider = ({ children }: TokenProviderProps) => {
  const [accessToken, setAccessToken] = useState<string>('');

  return (
    <TokenContext.Provider value={{ accessToken, setAccessToken }}>
      {children}
    </TokenContext.Provider>
  );
};

const useTokenContext = () => {
  const context = React.useContext(TokenContext);
  if (typeof context === 'undefined') {
    throw new Error('useLibraries must be used within a LibraryProvider');
  }
  return context;
};

export default useTokenContext;
