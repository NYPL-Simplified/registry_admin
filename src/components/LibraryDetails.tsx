import React, { useContext } from 'react';
import {
  Box,
  Form,
  FormField,
  FormRow,
  Table,
  Tabs,
} from '@nypl/design-system-react-components';

import LibraryDetailsGrid from './LibraryDetailsGrid';
import StageSelect from './StageSelect';
import { LibraryData } from './RegistryAdmin';
import { TokenContext, TokenContextValues } from '../context/tokenContext';
import { UPDATE_LIBRARY_STAGE } from '../constants';
import {
  LibrariesContext,
  LibrariesContextValues,
} from '../context/librariesContext';

interface LibraryDetailsProps {
  library: LibraryData;
}

export type LibraryStage = 'testing' | 'production' | 'cancelled' | '';

const LibraryDetails = ({ library }: LibraryDetailsProps) => {
  const {
    uuid,
    basic_info: basicInfo,
    urls_and_contact: urlsAndContact,
    areas,
    stages,
  } = library;

  const { accessToken } = useContext(TokenContext) as TokenContextValues;
  const { setUpdatedLibrary } = useContext(
    LibrariesContext
  ) as LibrariesContextValues;

  const getEmailData = () => {
    const emailData = [];
    for (const key in urlsAndContact) {
      if (key.split('_')[1] === 'email') {
        emailData.push([
          key.split('_').join(' '),
          urlsAndContact[key as keyof typeof urlsAndContact] ||
            'None available',
        ]);
      }
    }

    return emailData;
  };

  const libraryTabsData = [
    {
      label: 'Basic Information',
      content: <LibraryDetailsGrid content={basicInfo} />,
    },
    {
      label: 'Contact & URLs',
      content: <LibraryDetailsGrid content={urlsAndContact} />,
    },
    {
      label: 'Areas',
      content: <LibraryDetailsGrid content={areas} />,
    },
  ];

  const fetchUpdatedLibrary = () => {
    fetch(`${process.env.REGISTRY_API_DOMAIN}/libraries/${uuid}`, {
      headers: { Authorization: `Bearer ${accessToken}` },
    })
      .then((response) => {
        if (response.ok) {
          return response.json();
        } else {
          throw new Error('There was an issue fetching the updated library.');
        }
      })
      .then((response) => {
        // Setting the updatedLibrary in LibrariesContext triggers a useEffect
        // which updates the librariesInContext to include the updatedLibrary.
        setUpdatedLibrary(response);
      })
      .catch((err) => console.log(err));
  };

  const handleStageChange = (stage: string, value: string) => {
    // UPDATE_LIBRARY_STAGE endpoint expects the 'Library Stage' and the 'Registry
    // Stage', even if only one value has been updated.
    const body = new FormData();
    body.append('uuid', uuid);
    body.append(
      'Library Stage',
      stage === 'libraryStage' ? value : stages.library_stage
    );
    body.append(
      'Registry Stage',
      stage === 'registryStage' ? value : stages.registry_stage
    );

    fetch(UPDATE_LIBRARY_STAGE, {
      headers: { Authorization: `Bearer ${accessToken}` },
      method: 'POST',
      body,
    })
      .then((response) => {
        if (response.ok) {
          // If the POST request is successful, we'll follow it with a GET request
          // to fetch the library that was just updated.
          fetchUpdatedLibrary();
        } else {
          throw new Error('There was an issue updating this library.');
        }
      })
      .catch((err) => console.log(err));
  };

  return (
    <>
      <Form id={`registry-form-${uuid}`}>
        <FormRow>
          <FormField>
            <StageSelect
              uuid={uuid}
              stage='libraryStage'
              value={stages.library_stage}
              handleStageChange={handleStageChange}
            />
          </FormField>
          <FormField>
            <StageSelect
              uuid={uuid}
              stage='registryStage'
              value={stages.registry_stage}
              handleStageChange={handleStageChange}
            />
          </FormField>
        </FormRow>
      </Form>
      <Table
        columnHeaders={['', '']}
        tableData={getEmailData()}
        useRowHeaders
      />
      <Box paddingTop='m'>
        <Tabs id={`tabs-${uuid}`} tabsData={libraryTabsData} />
      </Box>
    </>
  );
};

export default LibraryDetails;
