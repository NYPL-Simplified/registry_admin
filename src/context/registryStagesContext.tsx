import React, { createContext, Dispatch, useState } from 'react';
import { LibraryStage } from '../components/LibraryDetails';

export type RegistryStagesContextValues = {
  registryStages: Record<string, LibraryStage>;
  setRegistryStages: Dispatch<
    React.SetStateAction<Record<string, LibraryStage>>
  >;
};

type RegistryStagesProviderProps = {
  children: React.ReactNode;
};

export const RegistryStagesContext =
  createContext<RegistryStagesContextValues | null>(null);

export const RegistryStagesProvider = ({
  children,
}: RegistryStagesProviderProps) => {
  const [registryStages, setRegistryStages] = useState<
    Record<string, LibraryStage>
  >({});

  console.log('registryStages -->', registryStages);

  return (
    <RegistryStagesContext.Provider
      value={{ registryStages, setRegistryStages }}
    >
      {children}
    </RegistryStagesContext.Provider>
  );
};
