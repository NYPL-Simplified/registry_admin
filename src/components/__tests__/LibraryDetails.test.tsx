import * as React from 'react';
import {
  fireEvent,
  render,
  screen,
  waitFor,
  within,
} from '@testing-library/react';
import renderer from 'react-test-renderer';

import mockLibraries from '../../../data/mockData';
import LibraryDetails from '../LibraryDetails';
import { TokenContext } from '../../context/tokenContext';
import { LibrariesContext } from '../../context/librariesContext';

const mockFormData = new FormData();
mockFormData.append('uuid', '1');
mockFormData.append('Library Stage', 'testing');
mockFormData.append('Registry Stage', 'production');

const setAccessTokenMock = jest.fn();
const setLibrariesMock = jest.fn();
const setUpdatedLibraryMock = jest.fn();

const renderLibraryDetailsWithContext = (
  accessToken: string,
  libraries = mockLibraries,
  setAccessToken = setAccessTokenMock,
  setLibraries = setLibrariesMock,
  setUpdatedLibrary = setUpdatedLibraryMock
) => {
  render(
    <TokenContext.Provider value={{ accessToken, setAccessToken }}>
      <LibrariesContext.Provider
        value={{ libraries, setLibraries, setUpdatedLibrary }}
      >
        <LibraryDetails library={libraries[0]} />
      </LibrariesContext.Provider>
    </TokenContext.Provider>
  );
};

const renderLibraryDetailsForSnapshot = (
  accessToken: string,
  libraries = mockLibraries,
  setAccessToken = setAccessTokenMock,
  setLibraries = setLibrariesMock,
  setUpdatedLibrary = setUpdatedLibraryMock
) => {
  return (
    <TokenContext.Provider value={{ accessToken, setAccessToken }}>
      <LibrariesContext.Provider
        value={{ libraries, setLibraries, setUpdatedLibrary }}
      >
        <LibraryDetails library={libraries[0]} />
      </LibrariesContext.Provider>
    </TokenContext.Provider>
  );
};

describe('LibraryRegistryPage', () => {
  beforeEach(() => {
    renderLibraryDetailsWithContext('mockAccessToken');
  });

  it('renders the library details correctly', () => {
    const selects = screen.getAllByRole('combobox');
    const libraryStageSelect = selects[0];
    const registryStageSelect = selects[1];

    expect(selects.length).toEqual(2);
    expect(libraryStageSelect).toHaveValue('production');
    expect(registryStageSelect).toHaveValue('production');

    const emailTable = screen.getByRole('table');
    const emailsForValidation = within(emailTable).getAllByRole('row');

    expect(emailsForValidation[1]).toHaveTextContent(/contact email/i);
    expect(emailsForValidation[2]).toHaveTextContent(/copyright email/i);
    expect(emailsForValidation[3]).toHaveTextContent(/help email/i);

    const tabButtons = screen.getAllByRole('tab');
    let tabPanel = screen.getByRole('tabpanel');

    expect(tabButtons.length).toEqual(3);
    expect(tabPanel).toBeInTheDocument();
    expect(tabPanel).toHaveTextContent(/number_of_patrons/i);

    const secondTab = tabButtons[1];
    const thirdTab = tabButtons[2];

    fireEvent.click(secondTab);

    tabPanel = screen.getByRole('tabpanel');

    expect(tabPanel).toBeInTheDocument();
    expect(tabPanel).toHaveTextContent(/web_url/i);

    fireEvent.click(thirdTab);

    tabPanel = screen.getByRole('tabpanel');

    expect(tabPanel).toBeInTheDocument();
    expect(tabPanel).toHaveTextContent(/focus/i);
  });

  it('makes the appropriate requests if the user changes the values of Library or Registry Stage', async () => {
    (global as any).fetch = jest
      .fn()
      .mockReturnValueOnce(
        Promise.resolve({
          ok: true,
          status: 200,
        })
      )
      .mockReturnValueOnce(
        Promise.resolve({
          json: () => Promise.resolve(mockLibraries[0]),
          ok: true,
          status: 200,
        })
      );

    const selects = screen.getAllByRole('combobox');
    const libraryStageSelect = selects[0];

    fireEvent.change(libraryStageSelect, { target: { value: 'testing' } });

    await waitFor(() =>
      expect(fetch).toHaveBeenCalledWith(process.env.UPDATE_LIBRARY_STAGE, {
        body: mockFormData,
        headers: { Authorization: 'Bearer mockAccessToken' },
        method: 'POST',
      })
    );

    await waitFor(() => {
      expect(fetch).toHaveBeenCalledWith(
        `${process.env.REGISTRY_API_DOMAIN}/libraries/1`,
        {
          headers: { Authorization: 'Bearer mockAccessToken' },
        }
      );
    });
  });
});

describe('LibraryDetails Snapshot', () => {
  it('renders the UI snapshot correctly', () => {
    const page = renderer
      .create(renderLibraryDetailsForSnapshot('mockAccessToken'))
      .toJSON();
    expect(page).toMatchSnapshot();
  });
});
