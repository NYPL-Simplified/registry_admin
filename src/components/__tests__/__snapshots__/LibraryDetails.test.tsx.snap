// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LibraryDetails Snapshot renders the UI snapshot correctly 1`] = `
Array [
  <form
    aria-label="form"
    className="css-1xdhyk6"
    id="registry-form-1"
  >
    <div
      className="css-fadkce"
      id="registry-form-1-parent"
    >
      <div
        className="css-5wru3y"
        id="registry-form-1-child0"
      >
        <div
          className="css-7ajbp2"
          id="registry-form-1-child0-grandchild0"
        >
          <div
            className="css-1u8qly9"
            id="libraryStage-1-wrapper"
          >
            <div
              className="css-0"
            >
              <div
                className="css-0"
              >
                <label
                  className="css-1xdhyk6"
                  htmlFor="libraryStage-1"
                  id="libraryStage-1-label"
                >
                  Library Stage
                </label>
              </div>
              <div
                className="chakra-select__wrapper css-42b2qy"
              >
                <select
                  className="chakra-select css-1ik61og"
                  disabled={false}
                  id="libraryStage-1"
                  name="libraryStage"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  value="production"
                >
                  <option
                    value="testing"
                  >
                    Testing
                  </option>
                  <option
                    value="production"
                  >
                    Production
                  </option>
                  <option
                    value="cancelled"
                  >
                    Cancelled
                  </option>
                </select>
                <div
                  className="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden={true}
                    className="chakra-icon chakra-select__icon css-1nreziu"
                    focusable={false}
                    id="libraryStage-1-icon"
                    role="presentation"
                    style={
                      Object {
                        "color": "currentColor",
                        "height": "1em",
                        "width": "1em",
                      }
                    }
                    viewBox="0 0 24 14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title>
                      arrow icon
                    </title>
                    <path
                      clipRule="evenodd"
                      d="M10.526 12.871.263 1.676 1.737.324 12 11.52 22.263.324l1.474 1.352L13.474 12.87a2 2 0 0 1-2.948 0z"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              aria-atomic={true}
              aria-live="polite"
              className="css-bl03f5"
              data-isinvalid={false}
              id="libraryStage-1-helperText"
            />
          </div>
        </div>
        <div
          className="css-7ajbp2"
          id="registry-form-1-child0-grandchild1"
        >
          <div
            className="css-1u8qly9"
            id="registryStage-1-wrapper"
          >
            <div
              className="css-0"
            >
              <div
                className="css-0"
              >
                <label
                  className="css-1xdhyk6"
                  htmlFor="registryStage-1"
                  id="registryStage-1-label"
                >
                  Registry Stage
                </label>
              </div>
              <div
                className="chakra-select__wrapper css-42b2qy"
              >
                <select
                  className="chakra-select css-1ik61og"
                  disabled={false}
                  id="registryStage-1"
                  name="registryStage"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  value="production"
                >
                  <option
                    value="testing"
                  >
                    Testing
                  </option>
                  <option
                    value="production"
                  >
                    Production
                  </option>
                  <option
                    value="cancelled"
                  >
                    Cancelled
                  </option>
                </select>
                <div
                  className="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden={true}
                    className="chakra-icon chakra-select__icon css-1nreziu"
                    focusable={false}
                    id="registryStage-1-icon"
                    role="presentation"
                    style={
                      Object {
                        "color": "currentColor",
                        "height": "1em",
                        "width": "1em",
                      }
                    }
                    viewBox="0 0 24 14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title>
                      arrow icon
                    </title>
                    <path
                      clipRule="evenodd"
                      d="M10.526 12.871.263 1.676 1.737.324 12 11.52 22.263.324l1.474 1.352L13.474 12.87a2 2 0 0 1-2.948 0z"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              aria-atomic={true}
              aria-live="polite"
              className="css-bl03f5"
              data-isinvalid={false}
              id="registryStage-1-helperText"
            />
          </div>
        </div>
      </div>
    </div>
  </form>,
  <table
    className="chakra-table css-1xdhyk6"
    role="table"
  >
    <thead
      className="css-0"
    >
      <tr
        className="css-0"
        role="row"
      >
        <th
          className="css-0"
          scope="col"
        />
        <th
          className="css-0"
          scope="col"
        />
      </tr>
    </thead>
    <tbody
      className="css-0"
    >
      <tr
        className="css-0"
        role="row"
      >
        <th
          className="css-0"
          scope="row"
        >
          <span />
          <span>
            contact email
          </span>
        </th>
        <td
          className="css-0"
          role="gridcell"
        >
          <span />
          <span>
            actonlibrary@actonlibrary.org
          </span>
        </td>
      </tr>
      <tr
        className="css-0"
        role="row"
      >
        <th
          className="css-0"
          scope="row"
        >
          <span />
          <span>
            copyright email
          </span>
        </th>
        <td
          className="css-0"
          role="gridcell"
        >
          <span />
          <span>
            actonlibrary@actonlibrary.org
          </span>
        </td>
      </tr>
      <tr
        className="css-0"
        role="row"
      >
        <th
          className="css-0"
          scope="row"
        >
          <span />
          <span>
            help email
          </span>
        </th>
        <td
          className="css-0"
          role="gridcell"
        >
          <span />
          <span>
            actonlibrary@actonlibrary.org
          </span>
        </td>
      </tr>
    </tbody>
  </table>,
  <div
    className="css-11z2e9m"
  >
    <div
      className="chakra-tabs css-1xdhyk6"
      id="tabs-1"
    >
      <div
        className="css-l3f8vc"
      >
        <button
          aria-label="Scroll tabs left"
          className="chakra-button css-1c0zt5w"
          id="tabs-previous-tabs-1"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="chakra-icon css-1vgifmd"
            focusable={false}
            id="tabs-previous-icon-tabs-1"
            role="img"
            viewBox="0 0 24 14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              previous arrow
            </title>
            <path
              clipRule="evenodd"
              d="M10.526 12.871.263 1.676 1.737.324 12 11.52 22.263.324l1.474 1.352L13.474 12.87a2 2 0 0 1-2.948 0z"
              fillRule="evenodd"
            />
          </svg>
        </button>
        <div
          className="css-0"
        >
          <div
            className="css-tjdjds"
          >
            <div
              aria-orientation="horizontal"
              className="chakra-tabs__tablist css-k008qs"
              onKeyDown={[Function]}
              role="tablist"
            >
              <button
                aria-controls="tabs-1--tabpanel--1"
                aria-selected={false}
                className="chakra-tabs__tab css-abpfcx"
                id="tabs-1--tab--1"
                onClick={[Function]}
                onFocus={[Function]}
                role="tab"
                tabIndex={-1}
                type="button"
              >
                Basic Information
              </button>
              <button
                aria-controls="tabs-1--tabpanel--1"
                aria-selected={false}
                className="chakra-tabs__tab css-abpfcx"
                id="tabs-1--tab--1"
                onClick={[Function]}
                onFocus={[Function]}
                role="tab"
                tabIndex={-1}
                type="button"
              >
                Contact & URLs
              </button>
              <button
                aria-controls="tabs-1--tabpanel--1"
                aria-selected={false}
                className="chakra-tabs__tab css-abpfcx"
                id="tabs-1--tab--1"
                onClick={[Function]}
                onFocus={[Function]}
                role="tab"
                tabIndex={-1}
                type="button"
              >
                Areas
              </button>
            </div>
          </div>
        </div>
        <button
          aria-label="Scroll tabs right"
          className="chakra-button css-rvrlhv"
          id="tabs-next-tabs-1"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="chakra-icon css-1vgifmd"
            focusable={false}
            id="tabs-next-icon-tabs-1"
            role="img"
            viewBox="0 0 24 14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              next arrow
            </title>
            <path
              clipRule="evenodd"
              d="M10.526 12.871.263 1.676 1.737.324 12 11.52 22.263.324l1.474 1.352L13.474 12.87a2 2 0 0 1-2.948 0z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        className="chakra-tabs__tab-panels css-8atqhb"
      >
        <div
          aria-labelledby="tabs-1--tab-0"
          className="chakra-tabs__tab-panel css-sjt5nk"
          hidden={false}
          id="tabs-1--tabpanel-0"
          role="tabpanel"
          tabIndex={0}
        >
          <section
            className="css-1xdhyk6"
          >
            <dl>
              <dt>
                description
              </dt>
              <dd>
                Serving Old Saybrook, CT
              </dd>
              <dt>
                internal_urn
              </dt>
              <dd>
                None available
              </dd>
              <dt>
                name
              </dt>
              <dd>
                Acton Public Library
              </dd>
              <dt>
                number_of_patrons
              </dt>
              <dd>
                9
              </dd>
              <dt>
                online_registration
              </dt>
              <dd>
                None available
              </dd>
              <dt>
                pls_id
              </dt>
              <dd>
                None available
              </dd>
              <dt>
                short_name
              </dt>
              <dd>
                CTACTN
              </dd>
              <dt>
                timestamp
              </dt>
              <dd>
                None available
              </dd>
            </dl>
          </section>
        </div>
        <div
          aria-labelledby="tabs-1--tab-1"
          className="chakra-tabs__tab-panel css-sjt5nk"
          hidden={true}
          id="tabs-1--tabpanel-1"
          role="tabpanel"
          tabIndex={0}
        />
        <div
          aria-labelledby="tabs-1--tab-2"
          className="chakra-tabs__tab-panel css-sjt5nk"
          hidden={true}
          id="tabs-1--tabpanel-2"
          role="tabpanel"
          tabIndex={0}
        />
      </div>
    </div>
  </div>,
]
`;
